gk/fib.c

```diff
@@ -1893,6 +1893,248 @@ l_list_gk_fib6(lua_State *l)
 	return 1;
 }
 
+#define CTYPE_STRUCT_ETHER_CACHE_PTR "struct ether_cache *"
+
+static void
+list_hash_table_neighbors(lua_State *l, struct neighbor_hash_table *neigh_ht)
+{
+	uint32_t next = 0;
+	const void *key;
+	void *data;
+	int32_t index;
+
+	static bool assigned_type_ether_cache = false;
+	static uint32_t correct_ctypeid_ether_cache;
+
+	void *cdata;
+
+	if (!assigned_type_ether_cache) {
+		correct_ctypeid_ether_cache = luaL_get_ctypeid(l,
+			CTYPE_STRUCT_ETHER_CACHE_PTR);
+		assigned_type_ether_cache = true;
+	}
+
+	index = rte_hash_iterate(neigh_ht->hash_table,
+		(void *)&key, &data, &next);
+	while (index >= 0) {
+		lua_pushvalue(l, 2);
+		lua_insert(l, 3);
+		cdata = luaL_pushcdata(l, correct_ctypeid_ether_cache,
+			sizeof(struct ether_cache *));
+		*(struct ether_cache **)cdata = &neigh_ht->cache_tbl[index];
+		lua_insert(l, 4);
+
+		if (lua_pcall(l, 2, 1, 0) != 0) {
```

Comment:
Since we are just raising the Lua error further here, we can drop the `if` statement and just rename `lua_pcall()` with `lua_call()`.